api/include/opentelemetry/trace/span_id.h

```diff
@@ -58,7 +58,11 @@ class SpanId final
   bool operator!=(const SpanId &that) const noexcept { return !(*this == that); }
 
   // Returns false if the SpanId is all zeros.
-  bool IsValid() const noexcept { return *this != SpanId(); }
+  bool IsValid() const noexcept
+  {
+    static_assert(kSize == 8, "update is needed if kSize is not 8");
+    return *reinterpret_cast<const uint64_t *>(&rep_) != 0ull;
```

Comment:
nice. Just wondering whether this would beat `memcmp`. Also we could do something similar for trace_id:

 `return (!(uint64_t *) (&rep))[0] == 0ull ) &&  ((uint64_t *) (&rep))[1] == 0ull ))`

Reply:
This should beat `memcmp` if the `memcmp` call is not inlined, as this just involves a memory read and an integer compare (probably just 2 instructions). The other places may also be improved like this. The other thing which is unique here is a default instance is constructed before comparison which is also eliminated in this PR. I'll revisit other `memcmp` in this file.